Adding a "global" not_intent parameter for slot mappings (#7931)
RasaHQ/rasa: changelog/261.improvement.md

> Added an optional `global_not_intent` parameter in forms.

Reviewer: we also need a deprecation changelog and a migration guide item.

Reviewer: can you please also create an entry in the migration guide for 2.5? 🙌🏻

> - In order to use it, please add in your `domain.yml` file the `global_not_intent` parameter with a value/list of values of your choice after the form's name.

Reviewer: I'd link to the documentation section as well

> - In order to use it, please add in your `domain.yml` file the `ignored_intents` parameter with a value/list of values of your choice after the form's name. Please see https://rasa.com/docs/rasa/forms for more information.

Reviewer: let's use relativ links please
```suggestion
- In order to use it, please add in your `domain.yml` file the `ignored_intents` parameter with a value/list of values of your choice after the form's name. Please see [Forms](forms.mdx) for more information.
```

> - This parameter will then update/add the `not_intent` key of all individual slots of a form with its specified value (or list of values).

Reviewer: Can we explain that in a more user-facing way and less coupled to the implementation? E.g.
```suggestion
- all slot mappings of a form won't apply if the intent of a message is specific in `ignored_intents`
```

> - In order to use it, please add in your `domain.yml` file the `ignored_intents` parameter with a value/list of values of your choice after the form's name. Please see [Forms](forms.mdx) for more information.
> - All slot mappings of a form won't apply if the intent of a message is specific in `ignored_intents`.
> - This can be used in case the user never wants to fill any slots of a form with the specified intent, eg. chitchat. 

Reviewer 2: ```suggestion
- To use it, add the `ignored_intents` parameter  in your `domain.yml` file after the forms name and provide a list of intents to ignore. Please see [Forms](forms.mdx) for more information.
- This can be used in case the user never wants to fill any slots of a form with the specified intent, eg. chitchat. 
```

Reviewer 2: Since it's more completely described in the docs, shortened the changelog a bit
